City-of-Helsinki/drupal-helfi-kymp
# UHF-7075: Add update hook for creating nodes from terms.

# [UHF-7075](https://helsinkisolutionoffice.atlassian.net/browse/UHF-7075)


Other PR: https://github.com/City-of-Helsinki/drupal-helfi-kymp/pull/461 makes Project district taxonomy obsolete. For project and district search we need District nodes to replace the Project district taxonomy terms. This  update hook was created in favor not to have create District nodes manually.

## What was done


* Add new update hook for creating District nodes from Project district taxonomy terms

## How to install

* Make sure your instance is up and running on correct branch.
  * `git pull origin UHF-7075-create-district-nodes-drush-command`
  * `make fresh`
* Run `make drush-cr`

## How to test


* [] Check that update hook` helfi_kymp_content_update_9001` runs. Check that there are no errors and new District nodes and translations are created only if those don't exist yet. (e.g. Jätkäsaari has fi and sv language version but en should be created when update hook runs)
* [x] Check that District nodes and translations work normally
* [x] Check that District nodes and translations are unpublished

## Designers review



## Other PRs


* https://github.com/City-of-Helsinki/drupal-helfi-kymp/pull/461
